File: DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/ModulesViewModels/ReportsModuleViewModel.vb
Imports System
Imports System.IO
Imports System.Configuration
Imports DevExpress.DemoData.Helpers
Imports DevExpress.DemoReports
Imports DevExpress.XtraReports
Imports XtraReportsDemos

Namespace ProductsDemo.Modules
    Public Class ReportsModuleViewModel
        Shared Sub New()
            PatchDemoReportsConnectionStrings()
        End Sub
        Public Overridable Property DocumentSource() As IReport

        Public Sub OnLoaded()
            UpdateReport()
        End Sub
        Public Sub OnUnloaded()
            DocumentSource.StopPageBuilding()
        End Sub
        Private Sub UpdateReport()
            DocumentSource = New XtraReportsDemos.MasterDetailReport.Report()
            DocumentSource.CreateDocument(True)
        End Sub
        Private Shared Sub PatchDemoReportsConnectionStrings()
        Dim path As String = DevExpress.Utils.FilesHelper.FindingFileName(AppDomain.CurrentDomain.BaseDirectory, "Data\nwind.mdb", False)
            ConnectionHelper.SetDataDirectory(System.IO.Path.GetDirectoryName(path))
        End Sub
    End Class
End Namespace

File: DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/ModulesViewModels/ReportsModuleViewModel.cs
using System;
using System.IO;
using System.Configuration;
using DevExpress.DemoData.Helpers;
using DevExpress.DemoReports;
using DevExpress.XtraReports;
using XtraReportsDemos;

namespace ProductsDemo.Modules {
    public class ReportsModuleViewModel {
        static ReportsModuleViewModel() {
            PatchDemoReportsConnectionStrings();
        }
        public virtual IReport DocumentSource { get; set; }

        public void OnLoaded() {
            UpdateReport();
        }
        public void OnUnloaded() {
            DocumentSource.StopPageBuilding();
        }
        void UpdateReport() {
            DocumentSource = new XtraReportsDemos.MasterDetailReport.Report();
            DocumentSource.CreateDocument(true);
        }
        static void PatchDemoReportsConnectionStrings() {
		string path = DevExpress.Utils.FilesHelper.FindingFileName(AppDomain.CurrentDomain.BaseDirectory, @"Data\nwind.mdb", false);
            ConnectionHelper.SetDataDirectory(Path.GetDirectoryName(path));
        }
    }
}
